src/string/check_permutation.rs

```diff
@@ -0,0 +1,40 @@
+//! Given two strings s1 and s2 consisting of lowercase letters.
+//!
+//! Determine whether the characters of one of the strings can be
+//! rearranged to become another string.
+
+pub fn check_permutation(s1: &str, s2: &str) -> bool {
+    if s1.len() != s2.len() {
+        return false;
+    }
+    s1.chars()
+        .all(|c| s1.matches(c).count() == s2.matches(c).count())
```

Comment:
What is the time complexity of this? $O(n^2)$? I think one can do it in a _linear_ time.

Reply:
With the help of a hash table to record the number of occurrences of a character and then compare the time complexity should be linear time.